### ISSUE TITLE
apt-get dist-upgrade breaks when SPL update coincides with a new kernel version

### ISSUE THREAD

**Issue by gitbisector** [zfsonlinux/debian-installers#6]
I turned on the dailies to get 0.6.4-1.1 and did an apt-get dist-upgrade after that. Bad idea. dkms built a new kernel module so I thought everything would be ok. System wouldn't boot after that for lack of matching zfs kernel module.

Recovery was swift though: Went back to snapshot before the upgrade, rolled back all the FS in the pool and redid the upgrade making sure to upgrade the kernel first. Turned on the dailies after a reboot.

**Comment by FransUrbo** [comment]
> I turned on the dailies to get 0.6.4-1.1
> 
> Didn't need to do that. It's in the released…

Could you provide a full log of the upgrade? I have myself upgraded both kernel and ZoL without issues, so something must have gone wrong and I can't tell what without seeing the full install log.

**Comment by gitbisector** [comment]
Ok, will do a reproduction and save the log (been cloning VMs tonight, so should be a minor step). Calling it a night now though, will do so next run.

**Comment by gitbisector** [comment]
Reran the sequence and it failed the same way: "Failed to load ZFS module stack" after reboot.

From the log (posted below as promised) seemed updating initramfs and grub after dkms compiles the kernel module for the new kernel+zfs version might be what is missing. 

To confirm that theory I ran the sequence once more and after `apt-get dist-upgrade` but before reboot I ran `update-initramfs -u && update-grub` and then there is no issue after reboot.

```
Reading package lists...
Building dependency tree...
Reading state information...
The following packages will be upgraded:
  bind9-host binutils dnsutils dpkg-dev file host krb5-locales libbind9-80
  libdns88 libdpkg-perl libevent-2.0-5 libfreetype6 libgssapi-krb5-2 libisc84
  libisccc80 libisccfg82 libk5crypto3 libkrb5-3 libkrb5support0 libldap-2.4-2
  liblwres80 libmagic1 libtasn1-3 libx11-6 libx11-data libxml2
  linux-headers-3.2.0-4-amd64 linux-headers-3.2.0-4-common
  linux-image-3.2.0-4-amd64 linux-libc-dev openssl spl spl-dkms tzdata
34 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 44.1 MB of archives.
After this operation, 9,589 kB of additional disk space will be used.
Do you want to continue [Y/n]? Get:1 http://security.debian.org/ wheezy/updates/main linux-image-3.2.0-4-amd64 amd64 3.2.68-1+deb7u1 [23.5 MB]
Get:2 http://archive.zfsonlinux.org/debian/ wheezy-daily/main spl amd64 0.6.4-1-3-wheezy [16.9 kB]
Get:3 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libevent-2.0-5 amd64 2.0.19-stable-3+deb7u1 [174 kB]
Get:4 http://archive.zfsonlinux.org/debian/ wheezy-daily/main spl-dkms all 0.6.4-1-3-wheezy [518 kB]
Get:5 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libk5crypto3 amd64 1.10.1+dfsg-5+deb7u3 [113 kB]
Get:6 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libgssapi-krb5-2 amd64 1.10.1+dfsg-5+deb7u3 [148 kB]
Get:7 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libkrb5-3 amd64 1.10.1+dfsg-5+deb7u3 [393 kB]
Get:8 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libkrb5support0 amd64 1.10.1+dfsg-5+deb7u3 [49.9 kB]
Get:9 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libldap-2.4-2 amd64 2.4.31-2 [243 kB]
Get:10 http://ftp.debian.org/debian/ wheezy-proposed-updates/main file amd64 5.11-2+deb7u8 [53.4 kB]
Get:11 http://security.debian.org/ wheezy/updates/main linux-headers-3.2.0-4-amd64 amd64 3.2.68-1+deb7u1 [630 kB]
Get:12 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libmagic1 amd64 5.11-2+deb7u8 [205 kB]
Get:13 http://security.debian.org/ wheezy/updates/main linux-headers-3.2.0-4-common amd64 3.2.68-1+deb7u1 [3,589 kB]
Get:14 http://security.debian.org/ wheezy/updates/main linux-libc-dev amd64 3.2.68-1+deb7u1 [839 kB]
Get:15 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libtasn1-3 amd64 2.13-2+deb7u2 [67.8 kB]
Get:16 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libxml2 amd64 2.8.0+dfsg1-7+wheezy4 [905 kB]
Get:17 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libfreetype6 amd64 2.4.9-1.1+deb7u1 [453 kB]
Get:18 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libx11-data all 2:1.5.0-1+deb7u2 [194 kB]
Get:19 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libx11-6 amd64 2:1.5.0-1+deb7u2 [901 kB]
Get:20 http://ftp.debian.org/debian/ wheezy-updates/main tzdata all 2015d-0+deb7u1 [414 kB]
Get:21 http://ftp.debian.org/debian/ wheezy-proposed-updates/main dnsutils amd64 1:9.8.4.dfsg.P1-6+nmu2+deb7u4 [166 kB]
Get:22 http://ftp.debian.org/debian/ wheezy-proposed-updates/main bind9-host amd64 1:9.8.4.dfsg.P1-6+nmu2+deb7u4 [73.6 kB]
Get:23 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libisc84 amd64 1:9.8.4.dfsg.P1-6+nmu2+deb7u4 [184 kB]
Get:24 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libdns88 amd64 1:9.8.4.dfsg.P1-6+nmu2+deb7u4 [752 kB]
Get:25 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libisccc80 amd64 1:9.8.4.dfsg.P1-6+nmu2+deb7u4 [36.2 kB]
Get:26 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libisccfg82 amd64 1:9.8.4.dfsg.P1-6+nmu2+deb7u4 [63.1 kB]
Get:27 http://ftp.debian.org/debian/ wheezy-proposed-updates/main liblwres80 amd64 1:9.8.4.dfsg.P1-6+nmu2+deb7u4 [55.5 kB]
Get:28 http://ftp.debian.org/debian/ wheezy-proposed-updates/main host all 1:9.8.4.dfsg.P1-6+nmu2+deb7u4 [20.9 kB]
Get:29 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libbind9-80 amd64 1:9.8.4.dfsg.P1-6+nmu2+deb7u4 [42.6 kB]
Get:30 http://ftp.debian.org/debian/ wheezy-proposed-updates/main krb5-locales all 1.10.1+dfsg-5+deb7u3 [1,503 kB]
Get:31 http://ftp.debian.org/debian/ wheezy-proposed-updates/main binutils amd64 2.22-8+deb7u2 [4,793 kB]
Get:32 http://ftp.debian.org/debian/ wheezy-proposed-updates/main dpkg-dev all 1.16.16 [1,362 kB]
Get:33 http://ftp.debian.org/debian/ wheezy-proposed-updates/main libdpkg-perl all 1.16.16 [963 kB]
Get:34 http://ftp.debian.org/debian/ wheezy-proposed-updates/main openssl amd64 1.0.1e-2+deb7u16 [701 kB]
Reading changelogs...
Extracting templates from packages: 88%
Extracting templates from packages: 100%
Preconfiguring packages ...
Fetched 44.1 MB in 40s (1,084 kB/s)
(Reading database ... 
(Reading database ... 5%
(Reading database ... 100%
(Reading database ... 39565 files and directories currently installed.)
Preparing to replace libevent-2.0-5:amd64 2.0.19-stable-3 (using .../libevent-2.0-5_2.0.19-stable-3+deb7u1_amd64.deb) ...
Unpacking replacement libevent-2.0-5:amd64 ...
Preparing to replace libk5crypto3:amd64 1.10.1+dfsg-5+deb7u2 (using .../libk5crypto3_1.10.1+dfsg-5+deb7u3_amd64.deb) ...
Unpacking replacement libk5crypto3:amd64 ...
Preparing to replace libgssapi-krb5-2:amd64 1.10.1+dfsg-5+deb7u2 (using .../libgssapi-krb5-2_1.10.1+dfsg-5+deb7u3_amd64.deb) ...
Unpacking replacement libgssapi-krb5-2:amd64 ...
Preparing to replace libkrb5-3:amd64 1.10.1+dfsg-5+deb7u2 (using .../libkrb5-3_1.10.1+dfsg-5+deb7u3_amd64.deb) ...
Unpacking replacement libkrb5-3:amd64 ...
Preparing to replace libkrb5support0:amd64 1.10.1+dfsg-5+deb7u2 (using .../libkrb5support0_1.10.1+dfsg-5+deb7u3_amd64.deb) ...
Unpacking replacement libkrb5support0:amd64 ...
Preparing to replace libldap-2.4-2:amd64 2.4.31-1+nmu2 (using .../libldap-2.4-2_2.4.31-2_amd64.deb) ...
Unpacking replacement libldap-2.4-2:amd64 ...
Preparing to replace file 5.11-2+deb7u6 (using .../file_5.11-2+deb7u8_amd64.deb) ...
Unpacking replacement file ...
Preparing to replace libmagic1:amd64 5.11-2+deb7u6 (using .../libmagic1_5.11-2+deb7u8_amd64.deb) ...
Unpacking replacement libmagic1:amd64 ...
Preparing to replace libtasn1-3:amd64 2.13-2+deb7u1 (using .../libtasn1-3_2.13-2+deb7u2_amd64.deb) ...
Unpacking replacement libtasn1-3:amd64 ...
Preparing to replace libxml2:amd64 2.8.0+dfsg1-7+wheezy2 (using .../libxml2_2.8.0+dfsg1-7+wheezy4_amd64.deb) ...
Unpacking replacement libxml2:amd64 ...
Preparing to replace libfreetype6:amd64 2.4.9-1.1 (using .../libfreetype6_2.4.9-1.1+deb7u1_amd64.deb) ...
Unpacking replacement libfreetype6:amd64 ...
Preparing to replace libx11-data 2:1.5.0-1+deb7u1 (using .../libx11-data_2%3a1.5.0-1+deb7u2_all.deb) ...
Unpacking replacement libx11-data ...
Preparing to replace libx11-6:amd64 2:1.5.0-1+deb7u1 (using .../libx11-6_2%3a1.5.0-1+deb7u2_amd64.deb) ...
Unpacking replacement libx11-6:amd64 ...
Preparing to replace linux-image-3.2.0-4-amd64 3.2.65-1 (using .../linux-image-3.2.0-4-amd64_3.2.68-1+deb7u1_amd64.deb) ...
Unpacking replacement linux-image-3.2.0-4-amd64 ...
Examining /etc/kernel/postrm.d .
run-parts: executing /etc/kernel/postrm.d/initramfs-tools 3.2.0-4-amd64 /boot/vmlinuz-3.2.0-4-amd64
run-parts: executing /etc/kernel/postrm.d/zz-update-grub 3.2.0-4-amd64 /boot/vmlinuz-3.2.0-4-amd64
Preparing to replace tzdata 2015b-0wheezy1 (using .../tzdata_2015d-0+deb7u1_all.deb) ...
Unpacking replacement tzdata ...
Processing triggers for man-db ...
Setting up tzdata (2015d-0+deb7u1) ...

Current default time zone: 'America/Los_Angeles'
Local time is now:      Wed Apr 29 18:59:30 PDT 2015.
Universal Time is now:  Thu Apr 30 01:59:30 UTC 2015.
Run 'dpkg-reconfigure tzdata' if you wish to change it.

(Reading database ... 
(Reading database ... 5%
(Reading database ... 100%
(Reading database ... 39565 files and directories currently installed.)
Preparing to replace dnsutils 1:9.8.4.dfsg.P1-6+nmu2+deb7u3 (using .../dnsutils_1%3a9.8.4.dfsg.P1-6+nmu2+deb7u4_amd64.deb) ...
Unpacking replacement dnsutils ...
Preparing to replace bind9-host 1:9.8.4.dfsg.P1-6+nmu2+deb7u3 (using .../bind9-host_1%3a9.8.4.dfsg.P1-6+nmu2+deb7u4_amd64.deb) ...
Unpacking replacement bind9-host ...
Preparing to replace libisc84 1:9.8.4.dfsg.P1-6+nmu2+deb7u3 (using .../libisc84_1%3a9.8.4.dfsg.P1-6+nmu2+deb7u4_amd64.deb) ...
Unpacking replacement libisc84 ...
Preparing to replace libdns88 1:9.8.4.dfsg.P1-6+nmu2+deb7u3 (using .../libdns88_1%3a9.8.4.dfsg.P1-6+nmu2+deb7u4_amd64.deb) ...
Unpacking replacement libdns88 ...
Preparing to replace libisccc80 1:9.8.4.dfsg.P1-6+nmu2+deb7u3 (using .../libisccc80_1%3a9.8.4.dfsg.P1-6+nmu2+deb7u4_amd64.deb) ...
Unpacking replacement libisccc80 ...
Preparing to replace libisccfg82 1:9.8.4.dfsg.P1-6+nmu2+deb7u3 (using .../libisccfg82_1%3a9.8.4.dfsg.P1-6+nmu2+deb7u4_amd64.deb) ...
Unpacking replacement libisccfg82 ...
Preparing to replace liblwres80 1:9.8.4.dfsg.P1-6+nmu2+deb7u3 (using .../liblwres80_1%3a9.8.4.dfsg.P1-6+nmu2+deb7u4_amd64.deb) ...
Unpacking replacement liblwres80 ...
Preparing to replace host 1:9.8.4.dfsg.P1-6+nmu2+deb7u3 (using .../host_1%3a9.8.4.dfsg.P1-6+nmu2+deb7u4_all.deb) ...
Unpacking replacement host ...
Preparing to replace libbind9-80 1:9.8.4.dfsg.P1-6+nmu2+deb7u3 (using .../libbind9-80_1%3a9.8.4.dfsg.P1-6+nmu2+deb7u4_amd64.deb) ...
Unpacking replacement libbind9-80 ...
Preparing to replace krb5-locales 1.10.1+dfsg-5+deb7u2 (using .../krb5-locales_1.10.1+dfsg-5+deb7u3_all.deb) ...
Unpacking replacement krb5-locales ...
Preparing to replace binutils 2.22-8 (using .../binutils_2.22-8+deb7u2_amd64.deb) ...
Unpacking replacement binutils ...
Preparing to replace dpkg-dev 1.16.15 (using .../dpkg-dev_1.16.16_all.deb) ...
Unpacking replacement dpkg-dev ...
Preparing to replace libdpkg-perl 1.16.15 (using .../libdpkg-perl_1.16.16_all.deb) ...
Unpacking replacement libdpkg-perl ...
Preparing to replace linux-headers-3.2.0-4-amd64 3.2.65-1 (using .../linux-headers-3.2.0-4-amd64_3.2.68-1+deb7u1_amd64.deb) ...
Unpacking replacement linux-headers-3.2.0-4-amd64 ...
Preparing to replace linux-headers-3.2.0-4-common 3.2.65-1 (using .../linux-headers-3.2.0-4-common_3.2.68-1+deb7u1_amd64.deb) ...
Unpacking replacement linux-headers-3.2.0-4-common ...
Preparing to replace linux-libc-dev:amd64 3.2.65-1 (using .../linux-libc-dev_3.2.68-1+deb7u1_amd64.deb) ...
Unpacking replacement linux-libc-dev:amd64 ...
Preparing to replace openssl 1.0.1e-2+deb7u13 (using .../openssl_1.0.1e-2+deb7u16_amd64.deb) ...
Unpacking replacement openssl ...
Preparing to replace spl 0.6.4-1-wheezy (using .../spl_0.6.4-1-3-wheezy_amd64.deb) ...
Unpacking replacement spl ...
Preparing to replace spl-dkms 0.6.4-1-wheezy (using .../spl-dkms_0.6.4-1-3-wheezy_all.deb) ...

-------- Uninstall Beginning --------
Module:  spl
Version: 0.6.4
Kernel:  3.2.0-4-amd64 (x86_64)
-------------------------------------

Status: Before uninstall, this module version was ACTIVE on this kernel.

spl.ko:
 - Uninstallation
   - Deleting from: /lib/modules/3.2.0-4-amd64/updates/dkms/
 - Original module
   - No original module was found for this module on this kernel.
   - Use the dkms install command to reinstall any previous module version.


splat.ko:
 - Uninstallation
   - Deleting from: /lib/modules/3.2.0-4-amd64/updates/dkms/
 - Original module
   - No original module was found for this module on this kernel.
   - Use the dkms install command to reinstall any previous module version.

depmod....

DKMS: uninstall completed.

------------------------------
Deleting module version: 0.6.4
completely from the DKMS tree.
------------------------------
Done.
Unpacking replacement spl-dkms ...
Processing triggers for man-db ...
Setting up libevent-2.0-5:amd64 (2.0.19-stable-3+deb7u1) ...
Setting up libkrb5support0:amd64 (1.10.1+dfsg-5+deb7u3) ...
Setting up libk5crypto3:amd64 (1.10.1+dfsg-5+deb7u3) ...
Setting up libkrb5-3:amd64 (1.10.1+dfsg-5+deb7u3) ...
Setting up libgssapi-krb5-2:amd64 (1.10.1+dfsg-5+deb7u3) ...
Setting up libldap-2.4-2:amd64 (2.4.31-2) ...
Setting up libmagic1:amd64 (5.11-2+deb7u8) ...
Setting up file (5.11-2+deb7u8) ...
Setting up libtasn1-3:amd64 (2.13-2+deb7u2) ...
Setting up libxml2:amd64 (2.8.0+dfsg1-7+wheezy4) ...
Setting up libfreetype6:amd64 (2.4.9-1.1+deb7u1) ...
Setting up libx11-data (2:1.5.0-1+deb7u2) ...
Setting up libx11-6:amd64 (2:1.5.0-1+deb7u2) ...
Setting up linux-image-3.2.0-4-amd64 (3.2.68-1+deb7u1) ...
Running depmod.
Examining /etc/kernel/postinst.d.
run-parts: executing /etc/kernel/postinst.d/dkms 3.2.0-4-amd64 /boot/vmlinuz-3.2.0-4-amd64
run-parts: executing /etc/kernel/postinst.d/initramfs-tools 3.2.0-4-amd64 /boot/vmlinuz-3.2.0-4-amd64
update-initramfs: Generating /boot/initrd.img-3.2.0-4-amd64
run-parts: executing /etc/kernel/postinst.d/zz-update-grub 3.2.0-4-amd64 /boot/vmlinuz-3.2.0-4-amd64
Generating grub configuration file ...
Found linux image: /boot/vmlinuz-3.2.0-4-amd64
Found initrd image: /boot/initrd.img-3.2.0-4-amd64
done
Setting up libisc84 (1:9.8.4.dfsg.P1-6+nmu2+deb7u4) ...
Setting up libdns88 (1:9.8.4.dfsg.P1-6+nmu2+deb7u4) ...
Setting up libisccc80 (1:9.8.4.dfsg.P1-6+nmu2+deb7u4) ...
Setting up libisccfg82 (1:9.8.4.dfsg.P1-6+nmu2+deb7u4) ...
Setting up libbind9-80 (1:9.8.4.dfsg.P1-6+nmu2+deb7u4) ...
Setting up liblwres80 (1:9.8.4.dfsg.P1-6+nmu2+deb7u4) ...
Setting up bind9-host (1:9.8.4.dfsg.P1-6+nmu2+deb7u4) ...
Setting up host (1:9.8.4.dfsg.P1-6+nmu2+deb7u4) ...
Setting up dnsutils (1:9.8.4.dfsg.P1-6+nmu2+deb7u4) ...
Setting up krb5-locales (1.10.1+dfsg-5+deb7u3) ...
Setting up binutils (2.22-8+deb7u2) ...
Setting up libdpkg-perl (1.16.16) ...
Setting up dpkg-dev (1.16.16) ...
Setting up linux-headers-3.2.0-4-common (3.2.68-1+deb7u1) ...
Setting up linux-headers-3.2.0-4-amd64 (3.2.68-1+deb7u1) ...
Examining /etc/kernel/header_postinst.d.
run-parts: executing /etc/kernel/header_postinst.d/dkms 3.2.0-4-amd64
Setting up linux-libc-dev:amd64 (3.2.68-1+deb7u1) ...
Setting up openssl (1.0.1e-2+deb7u16) ...
Setting up spl (0.6.4-1-3-wheezy) ...
Setting up spl-dkms (0.6.4-1-3-wheezy) ...
Loading new spl-0.6.4 DKMS files...
Building only for 3.2.0-4-amd64
Building initial module for 3.2.0-4-amd64
Done.

spl:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/3.2.0-4-amd64/updates/dkms/

splat.ko:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/3.2.0-4-amd64/updates/dkms/

Running the post_install script:

depmod....

DKMS: install completed.
root@debian :~# reboot
```

**Comment by FransUrbo** [comment]
As far as I can see, everything is working EXACTLY as planned.
1. Unpacks and installs the new kernel
2. Unpacks the new spl-dkms package
3. Removes the old spl-dkms
4. Sets up the new kernel, new grub.cfg and new initrd (without spl because it was just removed)
5. Sets up the new kernel headers
6. Sets up and builds the new spl + spl-dkms (successfully)

EXCEPT, dkms isn't triggering a <code>update-initramfs</code> after it completed the spl-dkms build :(. I've never seen or heard it doing this before. But this (but especially point three through six above) is why dkms is utter crap and should be outlawed! :)

Dkms is a good idea, but it's a super-crappy design and implementation..

I know I'm asking a lot of you, but if you can try to dig out WHY it didn't trigger an <code>update-initramfs</code> I would HUGELY appreciate it!

But from what I can see now, it's a bug/issue with dkms so I'm closing this (I _have_ said all along that one needs to review and monitor the upgrade procedure very closely and take "appropriate" options manually).

**Comment by gitbisector** [comment]
@FransUrbo, took on your challenge and figured out WHY it didn't trigger an `update-initramfs` for this particular scenario. 

/var/lib/dkms/spl/0.6.4/build/dkms.conf says 

```
REMAKE_INITRD="no"
```

whereas 
/var/lib/dkms/zfs/0.6.4/build/dkms.conf has

```
BUILD_DEPENDS[0]="spl"
...
REMAKE_INITRD="$(
  if [ -e /usr/share/initramfs-tools/hooks/zfs \
       -o -e /usr/share/dracut/modules.d/90zfs ]
  then
    echo -n yes
  else
    echo -n no
  fi
)"
```

BUILD_DEPENDS[] was introduced into dkms by @dajhorn just for zfs.
It forces the install order, but does not change which modules actually get rebuilt. 

In this last update ZFS was unchanged, only SPL got modified. Hence initrd did not get remade.

The code that generates dkms.conf for spl sits in pkg-spl repo where pull-requests cannot be done on the debian code much like pkg-zfs.

So here's a patch for pkg-spl on top of the `master/debian/jessie/0.6.4-1-3` tag

```
commit bad2ffc6c261c522c80f2f90398e0b6d2cf104cf
Author: bisector <gitbisector@gmail.com>
Date:   Sun May 3 13:14:11 2015 -0700

    Make REMAKE_INITRD conditional.

    Systems that run root on zfs need an updated initrd when SPL module gets rebuilt.
    Code is straight from pkg-zfs

diff --git a/debian/spl-dkms.dkms b/debian/spl-dkms.dkms
index a1b86cc..edaa01f 100644
--- a/debian/spl-dkms.dkms
+++ b/debian/spl-dkms.dkms
@@ -19,7 +19,15 @@ POST_INSTALL="cp
   ${dkms_tree}/${PACKAGE_NAME}/${PACKAGE_VERSION}/build/module/Module.symvers
   ${dkms_tree}/${PACKAGE_NAME}/${PACKAGE_VERSION}/${kernelver}/${arch}/
 "
-REMAKE_INITRD="no"
+REMAKE_INITRD="$(
+  if [ -e /usr/share/initramfs-tools/hooks/zfs \
+       -o -e /usr/share/dracut/modules.d/90zfs ]
+  then
+    echo -n yes
+  else
+    echo -n no
+  fi
+)"
 MAKE[0]="make"
 BUILT_MODULE_NAME[0]="spl"
 BUILT_MODULE_LOCATION[0]="module/spl/"
```

**Comment by FransUrbo** [comment]
> So here's a patch for pkg-spl on top of the master/debian/jessie/0.6.4-1-3 tag

AWSOME! Thanx, I'll fix that 'right away'...

**Comment by FransUrbo** [comment]
Fixed in <code>0.6.4-1b</code> (as well as dailies <code>0.6.4-1-4</code>.
